src/lib/config_parser.c

```diff
@@ -112,3 +112,101 @@ bool config_parse(const char *filename, parse_config_fn_t parse_config_fn, void
 exit:
 	return ret;
 }
+
+char *config_get_value(const char *filename, const char *section, const char *key)
```

Comment:
Instead of re implementing a parser for the config file, use functions in common. There are also inconsistencies between first implementation and the second. Are keys always case insensitive or case sensitive?

Reply:
what do you mean I'm reimplementing it? One function is to parse the whole file, the second function is to look for a specific key within a given section of a config file. It is a different use case. 🤔

Reply:
The second function is almost an exact copy of the first function. This is a big hint that there are code that could be reused. The major problem on doing that is that it's really hard on the long term to keep the behavior on both functions in sync. For example, every bug that we find on parsing the config file will probably be needed to be fixed twice. Any change in the format of the config file, error messages, etc are all duplicated.
The first function is a function that parses the config file and visit all config values found. The second function could use the first to parse the file and filter just the item that you need.

One restriction I would add here is that it's not very efficient to parse files, so if we are planning to get multiple values from the same file it's better to read it only once and to extract everything we need. Something similar to what @karthikprabhuvinod proposed on the other PR.

Anyway, I think that having a config_get_value() function is useful.

Reply:
ok, now I get what you mean, I agree there is code to be reused here

Reply:
In the end I decided it was better to just parse all file from the start instead of looking for particular keys in the config file later